crates/bevy_picking/src/hover.rs

```diff
@@ -275,3 +275,116 @@ fn merge_interaction_states(
         new_interaction_state.insert(*hovered_entity, new_interaction);
     }
 }
+
+/// A component that allows users to use regular Bevy change detection to determine when the pointer
+/// enters or leaves an entity. Users should insert this component on an entity to indicate interest
+/// in knowing about hover state changes.
+///
+/// This is similar to the old Bevy [`Interaction`] component, except that it only tracks hover
+/// state, not button presses or other interactions.
+///
+/// The component's boolean value will be `true` whenever the pointer is currently hovering over the
+/// entity, or any of the entity's children. This value is guaranteed to only be mutated when the
+/// pointer enters or leaves the entity, allowing Bevy change detection to be used efficiently.
+/// This is in contrast to the [`HoverMap`] resource, which is updated every frame.
+///
+/// Typically, a simple hoverable entity or widget will have this component added to it. More
+/// complex widgets can have this component added to each hoverable part.
+///
+/// The computational cost of keeping the `Hovering` components up to date is relatively cheap,
+/// and linear in the number of entities that have the `Hovering` component inserted.
+#[derive(Component, Copy, Clone, Default, Eq, PartialEq, Debug, Reflect)]
+#[reflect(Component, Default, PartialEq, Debug, Clone)]
+pub struct Hovering(pub bool);
+
+/// Uses [`HoverMap`] changes to update [`Hovering`] components.
+pub fn update_hovering_states(
+    hover_map: Option<Res<HoverMap>>,
+    mut hovers: Query<(Entity, &mut Hovering)>,
+    parent_query: Query<&ChildOf>,
+) {
+    // Don't bother collecting ancestors if there are no hovers.
+    if hovers.is_empty() {
+        return;
+    }
+
+    let Some(hover_map) = hover_map else { return };
+
+    // Set which contains the hovered for the current pointer entity and its ancestors. The capacity
+    // is based on the likely tree depth of the hierarchy, which is typically greater for UI (because
+    // of layout issues) than for 3D scenes. A depth of 32 is a reasonable upper bound for most use
+    // cases.
+    let mut hover_ancestors = EntityHashSet::with_capacity(32);
+    if let Some(map) = hover_map.get(&PointerId::Mouse) {
+        for (hovered_entity, _) in map.iter() {
+            hover_ancestors.insert(*hovered_entity);
+            hover_ancestors.extend(parent_query.iter_ancestors(*hovered_entity));
+        }
+    }
+
+    // For each hovered entity, it is considered "hovering" if it's in the set of hovered ancestors.
+    for (entity, mut hoverable) in hovers.iter_mut() {
+        let is_hovering = hover_ancestors.contains(&entity);
+        hoverable.set_if_neq(Hovering(is_hovering));
+    }
```

Comment:
Hovered state isn't strictly transitive from child to parent since a child node can be positioned outside of its parents bounds.

Reply:
I believe this is how the CSS `:hover` selector actually works: when you are hovering over a child element, you are technically also hovering over it's parent. I've tried to match that behavior, as far as I understand it.

Note that in CSS, the overflow property of the parent can affect whether the child is hoverable. However, the picking system should already handle this.

Reply:
Yeah thinking about it there are circumstances where it makes sense that way.